cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/InfoAction.hs

```diff
@@ -0,0 +1,407 @@
+      tempBaseAbsPath = makeTmpBaseAbsPath tempAbsPath
+
+  work <- H.createDirectoryIfMissing $ tempAbsPath' </> "work"
+
+  let sbe = ShelleyBasedEraConway
+      era = toCardanoEra sbe
+      cEra = AnyCardanoEra era
+      fastTestnetOptions = cardanoDefaultTestnetOptions
+        { cardanoEpochLength = 100
+        , cardanoSlotLength = 0.1
+        , cardanoNodeEra = cEra
+        }
+
+  testnetRuntime@TestnetRuntime
+    { testnetMagic
+    , poolNodes
+    , wallets
+    }
+    <- cardanoTestnet fastTestnetOptions conf
+
+  poolNode1 <- H.headM poolNodes
+
+  poolSprocket1 <- H.noteShow $ nodeSprocket $ poolRuntime poolNode1
+
+  execConfig <- H.mkExecConfig tempBaseAbsPath poolSprocket1
+
+  let socketName' = IO.sprocketName poolSprocket1
+      socketBase = IO.sprocketBase poolSprocket1 -- /tmp
+      socketPath = socketBase </> socketName'
+
+  H.note_ $ "Sprocket: " <> show poolSprocket1
+  H.note_ $ "Abs path: " <> tempAbsBasePath'
+  H.note_ $ "Socketpath: " <> socketPath
+  H.note_ $ "Foldblocks config file: " <> configurationFile testnetRuntime
+
+  gov <- H.createDirectoryIfMissing $ work </> "governance"
+  proposalAnchorFile <- H.note $ work </> gov </> "sample-proposal-anchor"
+  infoActionFp <- H.note $ work </> gov </> "info.action"
+
+  H.writeFile proposalAnchorFile "dummy anchor data"
+
+  proposalAnchorDataHash <- H.execCli' execConfig
+    [ "conway", "governance"
+    , "hash", "anchor-data", "--file-text", proposalAnchorFile
+    ]
+
+  let stakeVkeyFp = gov </> "stake.vkey"
+      stakeSKeyFp = gov </> "stake.skey"
+
+  _ <- P.cliStakeAddressKeyGen tempAbsPath'
+         $ P.KeyNames { P.verificationKeyFile = stakeVkeyFp
+                      , P.signingKeyFile = stakeSKeyFp
+                      }
+
+  let drepVkeyFp :: Int -> FilePath
+      drepVkeyFp n = tempAbsPath' </> "drep-keys" </> ("drep" <> show n) </> "drep.vkey"
+
+      drepSKeyFp :: Int -> FilePath
+      drepSKeyFp n = tempAbsPath' </> "drep-keys" </> ("drep" <> show n) </> "drep.skey"
+
+
+  -- Create Drep registration certificates
+  let drepCertFile :: Int -> FilePath
+      drepCertFile n = gov </> "drep-keys" <>"drep" <> show n <> ".regcert"
+  H.forConcurrently_ [1..3] $ \n -> do
+    H.execCli' execConfig
+       [ "conway", "governance", "drep", "registration-certificate"
+       , "--drep-verification-key-file", drepVkeyFp n
+       , "--key-reg-deposit-amt", show @Int 0
+       , "--out-file", drepCertFile n
+       ]
+
+  -- Retrieve UTxOs for registration submission
+
+  void $ H.execCli' execConfig
+    [ "conway", "query", "utxo"
+    , "--address", Text.unpack $ paymentKeyInfoAddr $ head wallets
+    , "--cardano-mode"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--out-file", work </> "utxo-1.json"
+    ]
+
+  txin1 <- do
+    utxoJson <- H.leftFailM . H.readJsonFile $ work </> "utxo-1.json"
+    UTxO utxo <- H.noteShowM $ decodeEraUTxO sbe utxoJson
+    H.noteShow =<< H.headM (Map.keys utxo)
```

Comment:
Could you make this a function? This is used a number of times in this test, but also in others.

Making it a function would reduce the test's length and make the important part stand out a bit more.